contracts/src/v0.8/dev/automation/2_1/KeeperRegistry2_1.sol

```diff
@@ -480,6 +495,52 @@ contract KeeperRegistry2_1 is KeeperRegistryBase2_1, OCR2Abstract, Chainable, ER
     return true;
   }
 
+  function _getTrigger(uint256 upkeepId) private pure returns (Trigger) {
+    // TODO
+    return Trigger.CONDITION;
+  }
+
+  /**
+   * @dev Does some early sanity checks before actually performing an upkeep
+   */
+  function _validateConditionalTrigger(
+    uint256 upkeepId,
+    ConditionalTrigger memory trigger,
+    Upkeep memory upkeep
+  ) internal returns (bool) {
+    if (trigger.blockNum < upkeep.lastPerformBlockNumber) {
+      // Can happen when another report performed this upkeep after this report was generated
+      emit StaleUpkeepReport(upkeepId);
```

Comment:
can we also add the trigger in the log

Reply:
what's the need? I think we're all pretty aligned on trying to encode the trigger type in the upkeepID, so it should be determinable from just the ID :)